### PR TITLE
Feature: implemented socketio rooms and joinroom to frontend and api

### PR DESCRIPTION
**Pull request by WallAshing**

### CODE REVIEW

File: api/src/main.js [hunk 1]
```diff
@@ -44,5 +47,68 @@ if (import.meta.env.PROD) {
   });
 }
 
+// ---------------------------------------------------------------------------------------------------------------------
```
Comment (oellan) [inline api/src/main.js:50]: Please delete useless comments

File: api/src/main.js [hunk 2]
```diff
@@ -44,5 +47,68 @@ if (import.meta.env.PROD) {
...
+
+const onConnection = (socket) => {
+  socketReceiveHandler(io, socket, users, roomExist, getRoomUserList, rooms);
+}
+
+io.on("connection", (socket) => {
+  onConnection(socket)
+})
+
+io.listen(3001);
```
Comment (oellan) [inline api/src/main.js:111]: Use the same port as the HTTP server.

Here is an example: https://socket.io/get-started/chat/

File: front/src/pages/rooms/RoomsPage.jsx [hunk 3]
```diff
@@ -29,15 +76,56 @@ export const RoomsPage = () => {
...
+      userUuid,
+      roomName,
+      roomDescription,
+      "1",
+      'https://meet.google.com/new'
+    )
+    // navigate(`/room/${roomName}`)
+  };
+
+  // const createRoom = async () => {
```
Comment (oellan) [inline front/src/pages/rooms/RoomsPage.jsx:93]: Remove useless comments

File: front/src/pages/rooms/RoomsPage.jsx [hunk 4]
```diff
@@ -29,15 +76,56 @@ export const RoomsPage = () => {
...
-              setShowModal(false)
-            }} />}
-        </section>
-    );
-}
+    <section className={RoomsPageStyle.page}>
+      <Header />
+      <h1 className={RoomsPageStyle.title}>Rooms</h1>
+      <Button text={"Create Room"} onClick={async () => {await handleClick()}}/>
+      <Button text={"Create Room 2"} onClick={async () => {await createRoom2("La room remplis de CACA", "roomDescription")}}/>
```
Comment (oellan) [inline front/src/pages/rooms/RoomsPage.jsx:124]: Remove the term "Caca" and replace with something else.

File: front/src/services/socketio/index.js [hunk 5]
```diff
@@ -10,15 +10,15 @@ import { io } from "socket.io-client";
 // }
 
 export const connectWS = () => {
-  return io.connect('http://127.0.0.1:3000/');
+  return io.connect(import.meta.env.VITE_WEB_SOCKET_API_URI);
```
Comment (oellan) [inline front/src/services/socketio/index.js:13]: Use the same port as the API

File: api/src/main.js (hunk 2 above)
Comment (WallAshing) [reply api/src/main.js:111]: Make it work and then we talk :)

File: api/src/main.js (hunk 1 above)
Comment (WallAshing) [reply api/src/main.js:50]: I don't see any emergency here, it is just to seperate clean code and no clean code

File: front/src/pages/rooms/RoomsPage.jsx (hunk 3 above)
Comment (WallAshing) [reply front/src/pages/rooms/RoomsPage.jsx:93]: CreateRoom comment removed ✅

File: front/src/services/socketio/index.js (hunk 5 above)
Comment (WallAshing) [reply front/src/services/socketio/index.js:13]: ie

File: front/src/pages/rooms/RoomsPage.jsx (hunk 4 above)
Comment (WallAshing) [reply front/src/pages/rooms/RoomsPage.jsx:124]: Done

File: api/src/main.js (hunk 2 above)
Comment (oellan) [reply api/src/main.js]: We can talk now

**Pull request merged**